draft-trammell-taps-interface.md

```diff
@@ -1033,16 +1033,12 @@ conditions holds:
 * the underlying protocol stack does not support message boundary
   preservation, and no deframer was supplied by the application
 
-In this case, the Message object passed to Received contains an indication
+In this case, the Message object passed to Received may contain an indication
 that the object received is partial, the byte offset of the data in the
 partial Message within the full Message, an indication whether this is the
 last (highest-offset) partial Message in the full Message, and an optional
 reference to the full Message it belongs to.
 
-Note that in the degenerate case -- no message boundary preservation and no
-deframing -- the entire connection is represented as one large message of
-indeterminate length.
-
```

Comment:
It seems important to leave this in (though on reading I can see that it might be useful to use less potentially-but-unintentionally-pejorative word than "degenerate" as an adjective for this case)

Reply:
It's not about the word "degenerate": it hints at "the FIN indicates the end of a message", which is a bad idea.

Reply:
If you insist on keeping this, please give better arguments than "it seems important" - because to me this text is confusing and potentially misleading.

Reply:
If you have a raw TCP connection with no framing protocol specified, how else do you know when the Message ends other than the end of the Connection? (i.e., the FIN indicates the end of message)

Note, though, that this is an extremely rare case, since you almost always have a framing protocol.

Reply:
This is connected to the discussion above: I now understand that you interpret this statement as "if you want anything better than the entire connection to be one huge message, you MUST provide a deframer".  So this is a way of saying "you really MUST provide a deframer." Now I think I also understand why Brian thought this is important. So, I think that's okay... I understand that this is a good idea because the transport system could then even apply this on the TCP byte stream (provided it has access to it) in order to cut out messages that arrive out of order. But then (for -01 ?) we should add such a flag to the receiver side to optionally offer, to do these Minion / Hollywood - like things.

Anyway. If my interpretation above is right, then I agree to keep this sentence.